src/framework/theme/components/autocomplete/autocomplete.component.spec.ts

```diff
@@ -0,0 +1,273 @@
+
+          <nb-option-group *ngFor="let group of filteredGroups" [title]="group.title">
+            <nb-option *ngFor="let option of group.options" [value]="option.value">
+              {{ option.title }}
+            </nb-option>
+          </nb-option-group>
+
+        </nb-autocomplete>
+
+      </nb-layout-column>
+    </nb-layout>
+  `,
+})
+export class NbAutocompleteTestComponent {
+
+  @Output() selectedChange: EventEmitter<any> = new EventEmitter();
+  @ViewChild(NbAutocompleteDirective, { static: true }) autocompleteDirective: NbAutocompleteDirective<string>;
+  @ViewChild(NbAutocompleteComponent, { static: false }) autocompletePanel: NbAutocompleteComponent<string>
+  @ViewChildren(NbOptionComponent) options: QueryList<NbOptionComponent<any>>;
+
+  @ViewChild('autoInput', { static: true }) input: HTMLInputElement;
+
+  groups = TEST_GROUPS;
+  filteredGroups;
+
+  constructor() {
+    this.filteredGroups = this.groups;
+  }
+
+  handleFunction(item) {
+    return item.value;
+  }
+
+  onChange($event) {
+    this.filteredGroups = this.filter(this.input.value);
+  }
+
+  onSelectionChange($event) {
+    this.filteredGroups = this.filter($event);
+  }
+
+  private filterOptions(options: any[], value: string) {
+    return options.filter(item => item.toLowerCase().includes(value));
+  }
+
+  private filter(value: string): any[] {
+    const filterValue = value.toLowerCase();
+    return this.groups
+      .map(group => {
+        return {
+          title: group.title,
+          options: this.filterOptions(group.options, filterValue),
+        }
+      })
+      .filter(group => group.options.length);
+    }
+}
+
+describe('Component: NbAutocompleteComponent', () => {
+  let fixture: ComponentFixture<NbAutocompleteTestComponent>;
+  let overlayContainerService: NbOverlayContainerAdapter;
+  let overlayContainer: HTMLElement;
+  let document: Document;
+  let input: HTMLInputElement;
+
+  const openPanel = () => {
+    fixture.componentInstance.autocompleteDirective.show();
+    fixture.detectChanges();
+  };
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        RouterTestingModule.withRoutes([]),
+        FormsModule,
+        ReactiveFormsModule,
+        NbThemeModule.forRoot(),
+        NbLayoutModule,
+        NbAutocompleteModule,
+        NbOptionListModule,
+      ],
+      declarations: [
+        NbAutocompleteTestComponent,
+      ],
+    });
+
+    fixture = TestBed.createComponent(NbAutocompleteTestComponent);
+    fixture.detectChanges();
+    input = fixture.debugElement.query(By.css('input')).nativeElement;
+    document = TestBed.get(NB_DOCUMENT);
+
+    overlayContainerService = TestBed.get(NbOverlayContainerAdapter);
+    overlayContainer = document.createElement('div');
+    overlayContainerService.setContainer(overlayContainer);
+
+    fixture.detectChanges();
+  });
+
+  afterEach(() => {
+    fixture.componentInstance.autocompleteDirective.hide();
```

Comment:
`fixture.componentInstance.autocompleteDirective` is used a lot, you can introduce `autocompleteDirective` variable and set it in beforeEach. This way `fixture.componentInstance.autocompleteDirective` will be shortened to `autocompleteDirective`

Reply:
Fixed